creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -274,15 +291,112 @@ class SnapbackSM {
+        primaryEndpoint: primary,
+        secondaryEndpoint: newSecondary
+      })
+    } catch (e) {
+      const errorMsg = `[issueUpdateReplicaSetOp] userId=${userId} wallet=${wallet} failed at phase=${phase} reconfiguring to spIds=[${newReplicaSetSPIds}]: ${e.toString()}\n${e.stack}`
+      this.logError(errorMsg)
+      throw new Error(errorMsg)
+    }
+  }
+
+  /**
+   * @param {Object} param
+   * @param {string[]} param.unhealthyReplicasSet an array of endpoints of unhealthy replica set nodes
+   * @param {string} param.primary current user's primary endpoint
+   * @param {string} param.secondary1 current user's first secondary endpoint
+   * @param {string} param.secondary2 current user's second secondary endpoint
+   * @returns {string} the endpoint of the newly selected secondary. Should not be a node in the
+   *        current replica set.
    */
-  async issueUpdateReplicaSetOp (userId, wallet, primary, secondary1, secondary2, unhealthyReplica) {
-    // await libs.userReplicaSetManagerClient._updateReplicaSet(userId, primary, [new secondaries], primary, [old secondaries])
+  async determineNewReplicaSet ({ primary, secondary1, secondary2, wallet }) {
+    // Fetch all the healthy nodes while disabling sync checks to select nodes for new replica set
+    if (!this.healthyNodes) {
+      const { services: healthyServicesMap } = await this.audiusLibs.ServiceProvider.autoSelectCreatorNodes({
+        performSyncCheck: false
+      })
+
+      this.healthyNodes = Object.keys(healthyServicesMap)
+      if (this.healthyNodes.length === 0) throw new Error('Unable to query any healthy Content Nodes')
+    }
+
+    // Make sure current replica set will not be chosen as potential nodes for the new replica set
+    const healthyPotentialPeers = this.healthyNodes.filter(node => node !== primary && node !== secondary1 && node !== secondary2)
+    let currentNodeHasExistingUserState = true
+    let i = 0
+    let newSecondary
+    while (currentNodeHasExistingUserState && i++ < MAX_SELECT_NEW_REPLICA_SET_ATTEMPTS) {
+      // Select a random node that is not from the current replica set
+      newSecondary = healthyPotentialPeers[Math.floor(Math.random() * healthyPotentialPeers.length)]
+
+      // Check to make sure that the newly selected secondary does not have state
+      const resp = await CreatorNode.getSyncStatus({
+        discoveryNodeEndpoint: this.audiusLibs.discoveryProvider.discoveryProviderEndpoint,
+        contentNodeEndpoint: newSecondary,
+        wallet
+      })
 
-    this.log(`Updating Replica Set for userID ${userId} & wallet ${wallet} from old replicaSet [${primary},${secondary1},${secondary2}] due to unhealthy replica ${unhealthyReplica}`)
+      if (resp.status.clockValue === -1) currentNodeHasExistingUserState = false
+    }
+
+    if (currentNodeHasExistingUserState) throw new Error(`Unable to select new secondary given current healthy Content Nodes=[${this.healthyNodes}]`)
+
+    return newSecondary
```

Comment:
returning a single new secondary is pretty restrictive here. i know it works in current case but very shortly after this, we will expand it. we should just return the full replicaset, as that's also more represenative of the functions intent "determineNewReplicaSet"

Reply:
will return entire new replica set